src/Numerics/ODESolvers/GenericCallbacks.jl

```diff
@@ -12,123 +12,111 @@ using ..ODESolvers
 This callback will run the function 'f()' every `time` wallclock time seconds.
 The `mpicomm` is used to syncronize runtime across MPI ranks.
 """
-struct EveryXWallTimeSeconds
+mutable struct EveryXWallTimeSeconds
     "time of the last callback"
-    lastcbtime_ns::Array{UInt64}
+    lastcbtime_ns::UInt64
     "time between callbacks"
     Δtime::Real
     "MPI communicator"
-    mpicomm
+    mpicomm::MPI.Comm
     "function to execute for callback"
-    func::Function
-    function EveryXWallTimeSeconds(func, Δtime, mpicomm)
-        lastcbtime_ns = [time_ns()]
-        new(lastcbtime_ns, Δtime, mpicomm, func)
+    func
+    "function to call to initialize the callback"
+    init
+    function EveryXWallTimeSeconds(func, Δtime, mpicomm, init = () -> nothing)
+        lastcbtime_ns = zero(UInt64)
+        new(lastcbtime_ns, Δtime, mpicomm, func, init)
     end
 end
-function (cb::EveryXWallTimeSeconds)(initialize::Bool = false)
-    # Is this an initialization call? If so, start the timers
-    if initialize
-        cb.lastcbtime_ns[1] = time_ns()
-        # If this is initialization init the callback too
-        try
-            cb.func(true)
-        catch
-        end
-        return 0
-    end
 
+function initialize!(cb::EveryXWallTimeSeconds, solver, Q, param, t0)
+    cb.lastcbtime_ns = time_ns()
+    cb.init()
```

Comment:
I'm not sure if we would want to pass `solver, Q, param, t` to `init` and the main `func` as well?

Reply:
It would be useful as it would allow us to move at least some of the callbacks to being functions rather than closures.

Reply:
okay, let me look into this.

Reply:
I came up with a really nice way to do this!